app/main.py

```diff
@@ -1,2 +1,103 @@
+from __future__ import annotations
+from typing import Any, Iterator
+
+
+class Node:
+    def __init__(self, key: Any, key_hash: int, value: Any) -> None:
+        self.key = key
+        self.key_hash = key_hash
+        self.value = value
+
+
 class Dictionary:
-    pass
+    def __init__(self, capacity: int = 8) -> None:
+        self.capacity = capacity
+        self.load_factor = 2 / 3
+        self.threshold = self.capacity * self.load_factor
+        self.nodes = [None] * self.capacity
+        self.size = 0
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        if self.size >= self.threshold:
+            self.resize()
+
+        key_hash = hash(key)
+        index = key_hash % self.capacity
+
+        while self.nodes[index]:
```

Comment:
This block of code with slight variations repeats in other methods - make it DRY